### ISSUE TITLE
Offer reduced performance mode to avoid high temperature on iPhone

### ISSUE THREAD

**Issue by nfsvsgta** [utmapp/UTM#2734]
As the title says, phone getting very hot while using any vm. Of course this is because of they high performance (using iPhone 12 Pro Max). Is there any way I can limit the power being consumed/ lower the cpu clock etc just to make utm take less resources and get less hot on the expense of performance?

**Comment by osy** [comment]
iOS should do this at a system level (i.e. it gets hot but never enough to physically harm you or the device). Apple's APIs doesn't provide an easy way to throttle resources, but you can always configure a VM with less RAM and with only a single core to reduce performance.

**Comment by nfsvsgta** [comment]
I agree, the iPhone is able to handle and deal with it accordingly. But I was thinking of a way to modify the cpu clock. I had a similar issue with dolphin where phone got hot while games ran at 60 fps. Setting cpu clock to 50% reduced the fps to 30 however phone no longer got hot. This also resulted in the battery not decreasing quick. I was thinking of something similar. The vm works incredibly fast but I was wondering of a way to lower cpu clock, a setting that maybe capped it to 50% etc, so I can preserve battery, phone dosent get hot etc.

**Comment by osy** [comment]
Hmm, that would have to be something that QEMU would have to implement. UTM only provides a UI to QEMU.

**Comment by conath** [comment]
@osy could we use [cpu_throttle_set](https://github.com/utmapp/qemu/blob/utm-edition/softmmu/cpu-throttle.c) for this?

**Comment by osy** [comment]
TIL
